GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
+                {
+                    uploadHandler = new UploadHandlerRaw(pData)
+                };
+            }
+            else
+                www = UnityWebRequest.Put(url, pData);
+
+            www.downloadHandler = new DownloadHandlerBuffer();
+            www.SetRequestHeader("Authorization", "Bearer " + api_access_token.access_token);
+            www.SetRequestHeader("Content-Type", "application/json");
+            yield return www.SendWebRequest();
+
+            string responseBody = www.downloadHandler.text;
+
+            StatisticResponse statistic = null;
+
+            if (www.isNetworkError || www.isHttpError)
+            {
+                Debug.LogWarning("Error submitting statistic: " + www.error);
+                Debug.Log("GlobalstatsIO API Response: " + responseBody);
+                callback(false);
+            }
+            else
+            {
+                statistic = JsonUtility.FromJson<StatisticResponse>(responseBody);
+            }
+
+            // ID is available only on create, not on update, so do not overwrite it
+            if (statistic._id != null && statistic._id != "")
+                statistic_id = statistic._id;
+
+            user_name = statistic.name;
+
+            //Store the returned data statically
+            foreach (StatisticValues value in statistic.values)
+            {
+                bool updated_existing = false;
+                for (int i = 0; i < statistic_values.Count; i++)
+                {
+                    if (statistic_values[i].key == value.key)
+                    {
+                        statistic_values[i] = value;
+                        updated_existing = true;
+                        break;
+                    }
+                }
+                if (!updated_existing)
+                {
+                    statistic_values.Add(value);
+                }
+            }
+
+            callback(true);
+        }
+
+        public StatisticValues GetStatistic(string key)
+        {
+            for (int i = 0; i < statistic_values.Count; i++)
+            {
+                if (statistic_values[i].key == key)
+                {
+                    return statistic_values[i];
+                }
+            }
+            return null;
+        }
+
+        public IEnumerator LinkStatistic(string id = "", Action<bool> callback = null)
+        {
+            if (api_access_token == null || !api_access_token.IsValid())
+            {
+                yield return GetAccessToken();
+            }
+
+            // If no id is supplied but we have one stored, reuse it.
+            if (id == "" && statistic_id != "")
+                id = statistic_id;
+
+            string url = "https://api.globalstats.io/v1/statisticlinks/" + id + "/request";
+
+            string json_payload = "{}";
+            byte[] pData = Encoding.UTF8.GetBytes(json_payload);
+
+            UnityWebRequest www = new UnityWebRequest(url, "POST")
```

Comment:
`UnityWebRequest` implements the dispose pattern.
You should make use of the `using` directive to cleanup the resources.
See: https://docs.unity3d.com/ScriptReference/Networking.UnityWebRequest.Post.html